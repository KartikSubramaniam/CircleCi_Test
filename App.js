import React from "react";
import { TouchableOpacity, Text, TextInput, View, StyleSheet } from "react-native";
import {RadioButton} from 'react-native-paper'

const App = () => {
  const [message, setMessage] = React.useState();
  const [checked, setChecked] = React.useState('first');

  return (
    <View style={{flex: 1}}>
      <TouchableOpacity 
        style={{marginTop: 10, height: 200, width: 200, backgroundColor: 'yellow', allignSelf:'center'}}
        onPress={() => {
          setTimeout(() => {
            setMessage("Hello Tester");
          }, Math.floor(Math.random() * 200));
        }}>
        <Text>
          Say Hello
        </Text>
      </TouchableOpacity>
      {message && (
        <Text style={styles.messageText} testID="printed-message">
          {message}
        </Text>
      )}
      <RadioButton
        testID="first"
        value="first"
        status={checked === 'first' ? 'checked' : 'unchecked'}
        onPress={() => setChecked('first')}
        color="red"
      />
      <RadioButton
        testID="second"
        value="second"
        status={checked === 'second' ? 'checked' : 'unchecked'}
        onPress={() => setChecked('second')}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  messageText: {
    fontFamily: "Arial",
    fontSize: 38,
    textAlign: "center",
    marginTop: 10
  }
});

export default App;