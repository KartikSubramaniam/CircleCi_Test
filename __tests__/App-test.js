/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { fireEvent, render, waitFor } from "@testing-library/react-native";


it('renders correctly', () => {
  renderer.create(<App />);
});

it("Renders Message", async () => {
  const { getByTestId, getByText, queryByTestId, toJSON } = render(<App />);

  const button = getByText("Say Hello");
  fireEvent.press(button);

  await waitFor(() => expect(queryByTestId("printed-message")).toBeTruthy());

  expect(getByTestId("printed-message").props.children).toBe("Hello Tester");
  expect(toJSON()).toMatchSnapshot();
});

it("Radio Button Test", async () => {
  const { getByTestId, getByText, queryByTestId, toJSON } = render(<App />);
  const radioButton = getByTestId("first")

  fireEvent.press(radioButton);
  expect(getByTestId("first").props.accessibilityState.checked).toBe(true)
  expect(getByTestId("second").props.accessibilityState.checked).toBe(false)

});