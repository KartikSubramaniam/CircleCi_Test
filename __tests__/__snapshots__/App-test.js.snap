// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Message 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "allignSelf": "center",
        "backgroundColor": "yellow",
        "height": 200,
        "marginTop": 10,
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Text>
      Say Hello
    </Text>
  </View>
  <Text
    style={
      Object {
        "fontFamily": "Arial",
        "fontSize": 38,
        "marginTop": 10,
        "textAlign": "center",
      }
    }
    testID="printed-message"
  >
    Hello Tester
  </Text>
  <View
    accessibilityLiveRegion="polite"
    accessibilityRole="radio"
    accessibilityState={
      Object {
        "checked": true,
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 18,
          "padding": 6,
        },
      ]
    }
    testID="first"
  >
    <View
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessibilityElementsHidden={true}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "color": "red",
              "fontSize": 24,
            },
          ]
        }
      >
        □
      </Text>
    </View>
  </View>
  <View
    accessibilityLiveRegion="polite"
    accessibilityRole="radio"
    accessibilityState={
      Object {
        "checked": false,
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 18,
          "padding": 6,
        },
      ]
    }
    testID="second"
  >
    <View
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <Text
        accessibilityElementsHidden={true}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "color": "#03dac4",
              "fontSize": 24,
            },
          ]
        }
      >
        □
      </Text>
    </View>
  </View>
</View>
`;
